sources/packages/api/src/services/notification/gc-notify-actions.service.ts

```diff
@@ -1,39 +1,77 @@
 import { Injectable } from "@nestjs/common";
-import { Student } from "../../database/entities";
 import { getExtendedDateFormat, getPSTPDTDateTime } from "../../utilities";
-import { STUDENT_FILE_UPLOAD_TEMPLATE_ID } from "../../utilities/system-configurations-constants";
-import { GCNotifyResult } from "./gc-notify.model";
+import {
+  MINISTRY_FILE_UPLOAD_TEMPLATE_ID,
+  STUDENT_FILE_UPLOAD_TEMPLATE_ID,
+} from "../../utilities/system-configurations-constants";
+import {
+  GCNotifyResult,
+  MinistryStudentFileUploadNotification,
+  MinistryStudentFileUploadPersonalisation,
+  StudentFileUploadNotification,
+  StudentFileUploadPersonalisation,
+} from "./gc-notify.model";
 import { GCNotifyService } from "./gc-notify.service";
 
 @Injectable()
 export class GCNotifyActionsService {
   constructor(private readonly gcNotifyService: GCNotifyService) {}
 
   /**
-   * This method is used to send email notification to SBC
+   * This method is used to send email notification to SABC
    * when a student uploads documents and submits it in the file uploader screen.
-   * @param student
-   * @param documentPurpose
-   * @param applicationNumber
+   * @param notification input parameters to generate the notification.
    * @returns GC Notify API call response.
    */
   async sendFileUploadNotification(
-    student: Student,
-    documentPurpose: string,
-    applicationNumber: string,
+    notification: StudentFileUploadNotification,
   ): Promise<GCNotifyResult> {
     const payload = {
-      email_address: this.gcNotifyService.gcNotifyToAddress(),
+      email_address: this.gcNotifyService.ministryToAddress(),
       template_id: STUDENT_FILE_UPLOAD_TEMPLATE_ID,
       personalisation: {
-        givenNames: student.user.firstName ?? "",
-        lastName: student.user.lastName,
-        dob: getExtendedDateFormat(student.birthDate),
-        applicationNumber: applicationNumber,
-        documentPurpose: documentPurpose,
-        date: `${getPSTPDTDateTime(new Date())} PST/PDT`,
+        givenNames: notification.firstName ?? "",
+        lastName: notification.lastName,
+        dob: getExtendedDateFormat(notification.birthDate),
+        applicationNumber: notification.applicationNumber,
+        documentPurpose: notification.documentPurpose,
+        date: this.getDateTimeOnBCTimeZone(),
```

Comment:
Why is the util method name as BCTimeZone instead of PSTTimeZone?

Reply:
Changed!